katello/katello-restore

```diff
@@ -75,14 +79,47 @@ def reset_katello
   puts "Done.\n"
 end
 
+def foreman_db_config(config_file = FOREMAN_DB_CONFIG)
+  config = YAML.load(File.read(config_file))
+  config['production']
+end
+
+def candlepin_db_config(config_file = CANDLEPIN_DB_CONFIG)
+  raw_config = File.read(config_file)
+  full_config = Hash[raw_config.scan(/(^[^#\n][^=]*)=(.*)/)]
+  uri = /:\/\/(([^\/:]*):?([^\/]*))\/([^?]*)(\?(.*))?/.match(full_config['org.quartz.dataSource.myDS.URL'])
+  @configuration = {
+      'username' => full_config['org.quartz.dataSource.myDS.user'],
+      'password' => full_config['org.quartz.dataSource.myDS.password'],
+      'database' => uri[4],
+      'host' => uri[2],
+      'port' => uri[3] || '5432'
+  }
+end
+
+def is_remote_db?(host)
+  host != 'localhost'
+end
+
+def run_pg_cmd(config, command, args)
+  run_cmd("PGPASSWORD='#{config[%{password}]}' #{command} -U #{config['username']} -h #{config['host'] || 'localhost'} -p #{config['port'] || '5432'} -d #{config['database']} #{args}")
+end
+
 def restore_psql_dumps
   puts "Restoring postgres dump files"
   run_cmd("katello-service start --only postgresql")
-  run_cmd("runuser - postgres -c 'dropdb foreman'")
-  run_cmd("runuser - postgres -c 'dropdb candlepin'")
-  run_cmd("runuser - postgres -c 'psql -f #{File.join(@dir, "pg_globals.dump")} postgres 2>/dev/null'")
-  run_cmd("runuser - postgres -c 'pg_restore -C -d postgres #{File.join(@dir, "foreman.dump")}'")
-  run_cmd("runuser - postgres -c 'pg_restore -C -d postgres #{File.join(@dir, "candlepin.dump")}'")
+  foreman_config = foreman_db_config
+  if is_remote_db?(foreman_config[:host])
+    candlepin_config = candlepin_db_config
+    run_pg_cmd(foreman_config, 'pg_restore', "--no-privileges --clean -n public #{File.join(@dir, "foreman.dump")}")
+    run_pg_cmd(candlepin_config, 'pg_restore', "--no-privileges --clean -n public #{File.join(@dir, "candlepin.dump")}")
```

Comment:
Is it possible it might be in a namespace other than `public` ?

Reply:
@sean797 I had to keep it restricted to the public schema. Otherwise it tried to clean also the procedural language and objects that the user may not be allowed to drop.